driver-core/src/main/com/mongodb/internal/connection/SaslAuthenticator.java

```diff
@@ -202,8 +217,8 @@ public void onResult(final BsonDocument result, final Throwable t) {
                 callback.onResult(null, wrapException(t));
                 disposeOfSaslClient(saslClient);
             } else if (result.getBoolean("done").getValue()) {
-                callback.onResult(null, null);
                 disposeOfSaslClient(saslClient);
+                verifySaslClientComplete(saslClient, callback);
```

Comment:
We shouldn't try to verify completeness after disposing of the SaslSlient.

Reply:
I have switched the two lines.